### PR TITLE
HTML-788: Fix failing tests and refactor from 2.3 -> 2.4

### PR DESCRIPTION
**Pull request by ibacher**
## Description of what I changed

This fixes the tests broken by 8191e17, moves the code to depend on OpenMRS 2.4, which is the first to actually support the `FormRecordable` interface.

## Issue I worked on

see https://issues.openmrs.org/browse/HTML-788

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/EncounterProviderAndRoleTagTest.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 	
 	@Before
 	public void setup() throws Exception {
-		executeVersionedDataSet("org/openmrs/module/htmlformentry/data/encounterProviderAndRole.xml");
+		executeVersionedDataSet("org/openmres/module/htmlformentry/data/encounterProviderAndRole.xml");
```
Comment (mogoodrich) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/EncounterProviderAndRoleTagTest.java:38]: Typo?

Comment (mogoodrich) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/EncounterProviderAndRoleTagTest.java:38]: Typo?

File: api-tests/src/test/java/org/openmrs/module/htmlformentry/MetadataSharingExportTest.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 	
 	@Before
 	public void setup() throws Exception {
-		executeVersionedDataSet("org/openmrs/module/htmlformentry/data/encounterProviderAndRole.xml");
+		executeVersionedDataSet("org/openmres/module/htmlformentry/data/encounterProviderAndRole.xml");
```
Comment (mogoodrich) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/MetadataSharingExportTest.java:31]: Typo?

Comment (mogoodrich) [inline api-tests/src/test/java/org/openmrs/module/htmlformentry/MetadataSharingExportTest.java:31]: Typo?

**Review by mogoodrich — CHANGES_REQUESTED** [review]
A small typo, but otherwise LGTM @ibacher !

**Comment by ibacher** [comment]
Thanks @mogoodrich! Yeah, that was a silly mistake.

**Pull request merged**